src/main/scala/com/mozilla/telemetry/streaming/EventsToAmplitude.scala

```diff
@@ -0,0 +1,224 @@
+    import pings.sparkSession.implicits._
+
+    pings.flatMap( v => {
+        try {
+          parsePing(Message.parseFrom(v.get(0).asInstanceOf[Array[Byte]]), sample)
+        } catch {
+          case _: Throwable if !raiseOnError => Array[String]()
+        }
+      }).as[String]
+  }
+
+  def storeStreamingEvents(spark: SparkSession, opts: Opts): Unit = {
+    val httpSink = new HttpSink(opts.url(), Map("api_key" -> opts.apiKey()))
+
+    val pings = spark
+      .readStream
+      .format("kafka")
+      .option("kafka.bootstrap.servers", opts.kafkaBroker())
+      .option("failOnDataLoss", opts.failOnDataLoss())
+      .option("kafka.max.partition.fetch.bytes", 8 * 1024 * 1024) // 8MB
+      .option("spark.streaming.kafka.consumer.cache.maxCapacity", kafkaCacheMaxCapacity)
+      .option("subscribe", kafkaTopic)
+      .option("startingOffsets", opts.startingOffsets())
+      .load()
+
+    getEvents(pings.select("value"), opts.sample(), opts.raiseOnError())
+      .writeStream
+      .queryName(queryName)
+      .foreach(httpSink)
+      .start()
+      .awaitTermination()
+  }
+
+  def storeBatchEvents(spark: SparkSession, opts: Opts): Unit = {
+    val fmt = format.DateTimeFormat.forPattern("yyyyMMdd")
+
+    val from = fmt.parseDateTime(opts.from())
+    val to = opts.to.get match {
+      case Some(t) => fmt.parseDateTime(t)
+      case _ => DateTime.now.minusDays(1)
+    }
+
+    implicit val sc = spark.sparkContext
+
+    for (offset <- 0 to Days.daysBetween(from, to).getDays) {
+      val currentDate = from.plusDays(offset).toString("yyyyMMdd")
+      val pings = com.mozilla.telemetry.heka.Dataset("telemetry")
+        .where("sourceName") {
+          case "telemetry" => true
+        }.where("docType") {
+          case docType if allowedDocTypes.contains(docType) => true
+        }.where("appName") {
+          case appName if allowedAppNames.contains(appName) => true
+        }.where("submissionDate") {
+          case date if date == currentDate => true
+        }.records(opts.fileLimit.get)
+        .map(m => Row(m.toByteArray))
+
+      val schema = StructType(List(
+          StructField("value", BinaryType, true)
+      ))
+
+      val pingsDataFrame = spark.createDataFrame(pings, schema)
+
+      getEvents(pingsDataFrame, opts.sample(), opts.raiseOnError())
+        .repartition(opts.maxParallelRequests())
+        .foreachPartition{it: Iterator[String] =>
+          val httpSink = new HttpSink(opts.url(), Map("api_key" -> opts.apiKey()))
+          it.foreach{ event =>
+            httpSink.process(event)
+            java.lang.Thread.sleep(opts.minDelay())
+          }
+        }
+    }
+
+    spark.stop()
+  }
+
+  def main(args: Array[String]): Unit = {
+    val opts = new Opts(args)
+
+    val spark = SparkSession.builder()
+      .appName(queryName)
+      .getOrCreate()
+
+    opts.kafkaBroker.get match {
+      case Some(_) => storeStreamingEvents(spark, opts)
```

Comment:
Slight naming nit: we're not actually writing out in these "store*" methods, correct? It seems like "send" or something like it would be more appropriate